homeassistant/components/tplink/sensor.py

```diff
@@ -188,52 +120,23 @@ def __init__(
         """Initialize the sensor."""
         super().__init__(device, coordinator, feature=feature, parent=parent)
         self._feature: Feature
+        feature_info = FEATURE_INFO.get(feature.id, {})
         self.entity_description = _description_for_feature(
-            SensorEntityDescription, feature, native_unit_of_measurement=feature.unit
+            SensorEntityDescription,
+            feature,
+            native_unit_of_measurement=feature.unit,
+            **feature_info,
         )
-
-        # TODO: define `options` if type==Choice
-        #  Requires the enum device class to be set. Cannot be combined with state_class or native_unit_of_measurement.
+        self._async_call_update_attrs()
 
     @callback
     def _async_update_attrs(self) -> None:
         """Update the entity's attributes."""
         value = self._feature.value
         if value is not None and self._feature.precision_hint is not None:
             value = round(cast(float, value), self._feature.precision_hint)
+        # today's consumption not available, when device was off all the day
+        # so show it as zero
+        if self._feature.id == ATTR_TODAY_ENERGY_KWH and value is None:
+            value = 0.0
```

Comment:
Perhaps we could move this into the library?

Reply:
That's removed and will need to release the associated library change